extension/src/extension.ts

```diff
@@ -249,3 +251,41 @@ export function activate(context: vscode.ExtensionContext): void {
 export function deactivate(): void {
   // any need for cleaning?
 }
+
+// /**
+//  * Synchronizes the anonymous user ID between a local file and global workspace state, if necessary, and then starts Telemetry.
+//  * The global workspace state is preserved over vscode updates, so it is more durable than local extension file.
+//  * The user ID should then be read by the language server or debug service when sending telemetry.
+//  */
+function startTelemetry(
+  context: vscode.ExtensionContext,
+  settings: Settings
+): void {
+  let newInstallation = false;
+  let userId: string = context.globalState.get(userIdStateKey) || "";
+  try {
+    const path = context.asAbsolutePath(userIdFile);
```

Comment:
Where the userId file located? Feels like I'm lacking some knowledge of how VS Code works :sweat_smile:

Reply:
In the extension folder (same as before, but in the VSCode way of finding the path 🙂)